src/Resource/GenericResource.php

```diff
@@ -244,4 +222,69 @@ protected function postUnserialize(array $data)
         $this->repoPath = array_pop($data);
         $this->path = array_pop($data);
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function pathGetChild($path, $relPath)
+    {
+        if (!$this->getRepository()) {
+            throw ResourceNotFoundException::forPath($path.'/'.$relPath);
+        }
+
+        return $this->getRepository()->get($path.'/'.$relPath);
```

Comment:
That's a ScruntizerCI problem. It detects dupplicated code if I do copy the methods, so it fails. Another solution would be to not implement `getChild`, `hasChild`, `listChildren` and `hasChildren` in LinkResource (throw an exception for instance) and only use the target path.